code/game/objects/buckling.dm

```diff
@@ -1,23 +1,30 @@
-
-
 /atom/movable
 	var/can_buckle = 0
 	var/buckle_lying = -1 //bed-like behaviour, forces mob.lying = buckle_lying if != -1 //except -1 actually means "rotate 90 degrees to the left" as it is used by 1*buckle_lying.
 	var/buckle_requires_restraints = 0 //require people to be handcuffed before being able to buckle. eg: pipes
-	var/mob/living/buckled_mob = null
 	var/buckle_offset = 0
+	var/list/mob/living/buckled_mobs = list()
+	var/max_buckled_mobs = 1
 
 
 //Interaction
 /atom/movable/attack_hand(mob/living/user)
 	. = ..()
-	if(can_buckle && buckled_mob)
-		return user_unbuckle_mob(user)
+	if(can_buckle && has_buckled_mobs())
+		if(buckled_mobs.len > 1)
+			var/unbuckled = input(user, "Who do you wish to unbuckle?","Unbuckle Who?") as null|mob in buckled_mobs
+			return user_unbuckle_mob(unbuckled,user)
+		else
+			return user_unbuckle_mob(buckled_mobs[1],user)
 
 /atom/movable/attack_robot(mob/living/user)
 	. = ..()
-	if(can_buckle && buckled_mob && Adjacent(user)) // attack_robot is called on all ranges, so the Adjacent check is needed
-		return user_unbuckle_mob(user)
+	if(can_buckle && has_buckled_mobs() && Adjacent(user)) // attack_robot is called on all ranges, so the Adjacent check is needed
+		if(buckled_mobs.len > 1)
+			var/unbuckled = input(user, "Who do you wish to unbuckle?","Unbuckle Who?") as null|mob in buckled_mobs
```

Comment:
What happens if they self-unbuckle before the user chooses them in the input list?

Reply:
i think there should be another check for that hold on...it may be in here https://github.com/tgstation/tgstation/blob/05ab481eacea861817d2b01231328d78ca820987/code/game/objects/buckling.dm#L77

its tg code but i think i have the proc. if not i may could add another check somewhere... but the list only pops up if theres more then one mob buckled to a thing.

Reply:
Tested. if they unbuckled before it just presents the popup window again with a new list. can be canceled.